dashboard/app/models/lesson_group.rb

```diff
@@ -92,6 +86,12 @@ def self.add_lesson_groups(raw_lesson_groups, script, new_suffix, editor_experim
     end
   end
 
+  def self.prevent_changing_stable_i18n_key(script, raw_lesson_group)
+    if script.is_stable && ScriptConstants.i18n?(script.name) && !(I18n.t "data.script.name.#{script.name}.lesson_groups").keys.map(&:to_s).include?(raw_lesson_group[:key])
```

Comment:
same as above: can use ... `#{raw_lesson_group[:key]}` instead of `.keys.map...include?` ?

Reply:
even better, would it be possible to DRY this up a bit? I think a [Policy Object](https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/lib/policies) might be appropriate here